Git-commit: 7be134eb691f6a54b267dbc321530ce0221a76b1
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 25 Nov 2022 15:51:06 +0100
Subject: [PATCH] Revert "net: broadcom: Fix BCMGENET Kconfig"
References: bsn#19
Patch-mainline: v5.10.156

This reverts commit fbb4e8e6dc7b38b3007354700f03c8ad2d9a2118 which is
commit 8d820bc9d12b8beebca836cceaf2bbe68216c2f8 upstream.

It causes runtime failures as reported by Naresh and Arnd writes:

	Greg, please just revert fbb4e8e6dc7b ("net: broadcom: Fix BCMGENET Kconfig")
	in stable/linux-5.10.y: it depends on e5f31552674e ("ethernet: fix
	PTP_1588_CLOCK dependencies"), which we probably don't want backported
	from 5.15 to 5.10.

So it should be reverted.

Reported-by: Naresh Kamboju <naresh.kamboju@linaro.org>
Reported-by: Arnd Bergmann <arnd@arndb.de>
Link: https://lore.kernel.org/r/CA+G9fYsXomPXcecPDzDydO3=i2qHDM2RTtGxr0p2YOS6=YcWng@mail.gmail.com
Cc: YueHaibing <yuehaibing@huawei.com>
Cc: Florian Fainelli <f.fainelli@broadcom.com>
Cc: Jakub Kicinski <kuba@kernel.org>
Cc: Sasha Levin <sashal@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/net/ethernet/broadcom/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/broadcom/Kconfig b/drivers/net/ethernet/broadcom/Kconfig
index 2b6d929d462f..7b79528d6eed 100644
--- a/drivers/net/ethernet/broadcom/Kconfig
+++ b/drivers/net/ethernet/broadcom/Kconfig
@@ -69,7 +69,7 @@ config BCMGENET
 	select BCM7XXX_PHY
 	select MDIO_BCM_UNIMAC
 	select DIMLIB
-	select BROADCOM_PHY if (ARCH_BCM2835 && PTP_1588_CLOCK_OPTIONAL)
+	select BROADCOM_PHY if ARCH_BCM2835
 	help
 	  This driver supports the built-in Ethernet MACs found in the
 	  Broadcom BCM7xxx Set Top Box family chipset.
-- 
2.35.3

